src/utils.ts

```diff
@@ -31,22 +31,27 @@ export const isDOMElement = (element: unknown): boolean =>
 
 export const createNotifier =
   (
+    onResize: Props['onResize'],
+    sizeRef: Readonly<React.MutableRefObject<ReactResizeDetectorDimensions>>,
```

Comment:
Do we need this additional variable? Can't we simply compare `width` and `height` to the current state?

Reply:
I stored the current state inside a ref so that we can access its value within the effect without having to declare it as a dependency and recreate the ResizeObserver every time it changes. It's kind of a hacky approach to avoid stale values or triggering the effect too often

Or did you mean using the `setState((prev) => { ... });` pattern?

Reply:
I see. That's what you initially did and then refactored to the current solution...

Thank you! I'm going to merge the PR